packages/admin-panel/src/authentication/actions.js

```diff
@@ -61,9 +59,14 @@ export const loginError = errorMessage => ({
   errorMessage,
 });
 
-export const logout = () => ({
-  type: LOGOUT,
-});
+export const logout = () => async (dispatch, getState, { api }) => {
+  await api.logout();
+
+  dispatch({
+    // Set state to logging out
+    type: LOGOUT,
+  });
+};
```

Comment:
Perhaps dispatch the logout before waiting for the actual logout to complete, so that the user experiences it quickly. The ideal thing would be to add error handling, but I'm happy to just go with showing the logged out situation immediately